### PR TITLE
feat(backend): export certificate

### PR DESCRIPTION
**Pull request by emjunior258**
Implements the following API operations:

- `getCertificatePem`
- `getCertificateDer`

### CODE REVIEW

File: certeasy-backend-app/src/main/java/org/certeasy/backend/common/BaseResource.java [hunk 1]
```diff
@@ -46,10 +47,10 @@ public Response checkIssuerNotExistsThen(String issuerId, Supplier<Response> ope
             return optionalResponse.get();
         if(registry.exists(issuerId)){
             LOGGER.warn("Issuer id taken: " +  issuerId);
-            return Response.status(409).entity(
-                            new Problem("/problems/issuerId/id-taken",
+            return Response.status(409).entity(new Problem("/problems/issuerId/id-taken",
                                     "Issuer ID Taken", 409,
```
Comment (snhane) [inline certeasy-backend-app/src/main/java/org/certeasy/backend/common/BaseResource.java:51]: `return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(new Problem("/problems/issuerId/id-taken",
                                    "Issuer ID Taken", Response.Status.CONFLICT.getStatusCode(),`

File: certeasy-backend-app/src/main/java/org/certeasy/backend/common/BaseResource.java [hunk 2]
```diff
@@ -62,10 +63,11 @@ public Response checkIssuerExistsThen(String issuerId, IssuerOperation operation
...
         if(issuer.isEmpty()){
             LOGGER.warn("Issuer not found: " +  issuerId);
-            return Response.status(409).entity(
-                            new Problem("/problems/issuer/not-found",
+            return Response.status(404).entity(
+                    new Problem("/problems/issuer/not-found",
                                     "Issuer not found", 404,
                                     "There no issuer with a matching ID: " + issuerId))
+                    .type(MediaType.APPLICATION_JSON_TYPE)
                     .build();
```
Comment (snhane) [inline certeasy-backend-app/src/main/java/org/certeasy/backend/common/BaseResource.java:71]: @emjunior258, what if we replace the code number with the actual status?
`return Response.status(Response.Status.NOT_FOUND.getStatusCode()).entity(
                    new Problem("/problems/issuer/not-found",
                                    "Issuer not found", Response.Status.NOT_FOUND.getStatusCode(),
                                    "There no issuer with a matching ID: " + issuerId))
                    .type(MediaType.APPLICATION_JSON_TYPE)
                    .build();`

File: certeasy-backend-app/src/main/java/org/certeasy/backend/common/cert/NotFoundProblem.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.certeasy.backend.common.cert;
+
+
+import org.certeasy.backend.common.problem.Problem;
+
+public class NotFoundProblem extends Problem {
+
+    public NotFoundProblem(String type, String title, String detail) {
+        super(type, title, 404, detail);
```
Comment (snhane) [inline certeasy-backend-app/src/main/java/org/certeasy/backend/common/cert/NotFoundProblem.java:9]: replace 404 with `Response.Status.NOT_FOUND.getStatusCode()`?

**Review by snhane — CHANGES_REQUESTED** [review]
made some suggestions for some http response codes.

File: certeasy-backend-app/src/main/java/org/certeasy/backend/common/cert/NotFoundProblem.java (hunk 3 above)
Comment (emjunior258) [reply certeasy-backend-app/src/main/java/org/certeasy/backend/common/cert/NotFoundProblem.java:9]: I just think `Response.Status.NOT_FOUND.getStatusCode()` is more verbose and less readable. I mean, the status code for NOT_FOUND will never change. We are just writing more code. Also, the enum `Response.Status` doesn't cover all the status we are using, for example: `422`.

**Pull request merged**